dbms/src/AggregateFunctions/AggregateFunctionSum.cpp

```diff
@@ -13,6 +13,19 @@ AggregateFunctionPtr createAggregateFunctionSum(const std::string & name, const
     if (argument_types.size() != 1)
         throw Exception("Incorrect number of arguments for aggregate function " + name, ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
 
+    AggregateFunctionPtr res(createWithNumericTypeNearest<AggregateFunctionSum>(*argument_types[0]));
```

Comment:
This just instantiates a different variant of the template with `AggregateFunctionSum<T, NearestFieldType<T>::Type>`.